testing/unit/events_test.cpp

```diff
@@ -0,0 +1,133 @@
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#include <opae/fpga.h>
+
+#ifdef __cplusplus
+
+extern "C" {
+#endif
+#include "fpgad/config_int.h"
+#include "fpgad/log.h"
+#include "fpgad/srv.h"
+
+#ifdef __cplusplus
+}
+#endif
+#include <stdlib.h>
+#include <chrono>
+#include <thread>
+#include "gtest/gtest.h"
+#include "test_system.h"
+
+using namespace opae::testing;
+
+class events_p : public ::testing::TestWithParam<std::string> {
+ protected:
+  events_p()
+      : tmpsysfs_("mocksys-XXXXXX"),
+        tmpfpgad_log_("tmpfpgad-XXXXXX.log"),
+        tmpfpgad_pid_("tmpfpgad-XXXXXX.pid"),
+        handle_(nullptr) {}
+
+  virtual void SetUp() override {
+    tmpfpgad_log_ = mkstemp(const_cast<char *>(tmpfpgad_log_.c_str()));
+    tmpfpgad_pid_ = mkstemp(const_cast<char *>(tmpfpgad_pid_.c_str()));
+    std::string platform_key = GetParam();
+    ASSERT_TRUE(test_platform::exists(platform_key));
+    platform_ = test_platform::get(platform_key);
+    system_ = test_system::instance();
+    system_->initialize();
+    tmpsysfs_ = system_->prepare_syfs(platform_);
+
+    ASSERT_EQ(fpgaGetProperties(nullptr, &filter_), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_ACCELERATOR), FPGA_OK);
+    ASSERT_EQ(fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
+                            &num_matches_),
+              FPGA_OK);
+    ASSERT_EQ(fpgaOpen(tokens_[0], &handle_, 0), FPGA_OK);
+    ASSERT_EQ(fpgaCreateEventHandle(&eh_), FPGA_OK);
+    config_ = {
+        .verbosity = 0,
+        .poll_interval_usec = 100 * 1000,
+        .daemon = 0,
+        .directory = ".",
+        .logfile = tmpfpgad_log_.c_str(),
+        .pidfile = tmpfpgad_pid_.c_str(),
+        .filemode = 0,
+        .running = true,
+        .socket = "/tmp/fpga_event_socket",
+        .null_gbs = {0},
+        .num_null_gbs = 0,
+    };
+    open_log(tmpfpgad_log_.c_str());
+    fpgad_ = std::thread(server_thread, &config_);
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+  }
+
+  virtual void TearDown() override {
+    config_.running = false;
+    EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
+    EXPECT_EQ(fpgaDestroyEventHandle(&eh_), FPGA_OK);
+    if (handle_ != nullptr) EXPECT_EQ(fpgaClose(handle_), FPGA_OK);
+    if (!tmpsysfs_.empty() && tmpsysfs_.size() > 1) {
+      std::string cmd = "rm -rf " + tmpsysfs_;
+      std::system(cmd.c_str());
+    }
+    system_->finalize();
+    fpgad_.join();
+  }
+
+  std::string tmpsysfs_;
+  std::string tmpfpgad_log_;
+  std::string tmpfpgad_pid_;
+  struct config config_;
+  fpga_properties filter_;
+  std::array<fpga_token, 2> tokens_;
+  fpga_handle handle_;
+  uint32_t num_matches_;
+  test_platform platform_;
+  test_system *system_;
+  std::thread fpgad_;
+  fpga_event_handle eh_;
+};
+
+TEST_P(events_p, register_event) {
```

Comment:
Seems like we had a few more tests.  Adding them later?

Reply:
yes